Mathlib/MeasureTheory/Integral/CircleAverage.lean

```diff
@@ -0,0 +1,144 @@
+/-
+Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stefan Kebekus
+-/
+import Mathlib.MeasureTheory.Integral.CircleIntegral
+import Mathlib.MeasureTheory.Integral.IntervalAverage
+import Mathlib.MeasureTheory.Integral.IntervalIntegral.Periodic
+
+/-!
+# Circle Averages
+
+For a function `f` on the complex plane, this file introduces the notation `circleAverage f c R` as
+a shorthand for the average of `f` on the circle with center `c` and radius `R`, equipped with the
+rotation-invariant measure of total volume one. Like `IntervalAverage`, this notion exists as a
+convenience. It avoids notationally inconvenient compositions of `f` with `circleMap` and avoids the
+need to manually elemininate `2 * π` every time an average is computed.
+
+Note: Like the interval average defined in `Mathlib.MeasureTheory.Integral.IntervalAverage`, the
+`circleAverage` defined here is a purely measure-theoretic average. It should not be confused with
+`CircleIntegral`, which is the path integral over the circle path. The relevant integrability
+property `circleAverage` is `CircleIntegrable`, as defined in
+`Mathlib.MeasureTheory.Integral.CircleIntegral`.
+
+Implementation Note: Like `circleMap`, `circleAverage`s are defined for negative radii. The theorem
+`circleAverage_congr_negRadius` shows that the average is independent of the radius' sign.
+-/
+
+open Filter Metric Real
+
+variable
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+  {𝕜 : Type*} [NontriviallyNormedField 𝕜] [NormedSpace 𝕜 E] [SMulCommClass ℝ 𝕜 E]
+  {f f₁ f₂ : ℂ → E} {c : ℂ} {R : ℝ} {a : 𝕜}
+
```

Comment:
Could you move everything into the `Real` namespace? In this way, one could also define `Real.circleIntegral` without clashing with the complex notion.

Reply:
Implemented.